openstack/clustering/v1/policies/results.go

```diff
@@ -72,3 +74,62 @@ func (r *Policy) UnmarshalJSON(b []byte) error {
 
 	return nil
 }
+
+// Create Policy API returns time in RFC3339 with Z
+const RFC3339WithZ = "2006-01-02T15:04:05Z"
+
+type JSONRFC3339WithZ time.Time
+
+func (jt *JSONRFC3339WithZ) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+	if s == "" {
+		return nil
+	}
+	t, err := time.Parse(RFC3339WithZ, s)
+	if err != nil {
+		return err
+	}
+	*jt = JSONRFC3339WithZ(t)
+	return nil
+}
+
+func (r *PolicyFromCreate) UnmarshalJSON(b []byte) error {
```

Comment:
Instead of doing this, I came up with the following:

```go
func (r *Policy) UnmarshalJSON(b []byte) error {
  type tmp Policy
  var s struct {
    tmp
    CreatedAt string `json:"created_at,omitempty"`
    UpdatedAt string `json:"updated_at,omitempty"`
  }
  err := json.Unmarshal(b, &s)
  if err != nil {
    return err
  }
  *r = Policy(s.tmp)

  if s.CreatedAt != "" {
    r.CreatedAt, err = time.Parse(gophercloud.RFC3339MilliNoZ, s.CreatedAt)
    if err != nil {
      r.CreatedAt, err = time.Parse(RFC3339WithZ, s.CreatedAt)
      if err != nil {
        return err
      }
    }
  }

  if s.UpdatedAt != "" {
    r.UpdatedAt, err = time.Parse(gophercloud.RFC3339MilliNoZ, s.UpdatedAt)
    if err != nil {
      r.UpdatedAt, err = time.Parse(RFC3339WithZ, s.UpdatedAt)
      if err != nil {
        return err
      }
    }
  }

  return nil
}
```

This allows the same `Policy` result to be used, but will parse two different time formats.

@jrperritt is there a cleaner way of doing this?

Reply:
Done.